### ISSUE TITLE
Breaks after Home Assistant update 2023-5.0

### ISSUE THREAD

**Issue by rrediske** [ollo69/ha-smartthinq-sensors#535]
**Describe the bug**
When I update to Home Assistant Core 2023.5.0, I got errors that I reported to HA:
https://github.com/home-assistant/core/issues/92530
My last attempt, I updated pip and cchardet and then this repo, and I get the errors listed below

**Expected behavior**
It worked great before, expected the same

**Screenshots**
Nothing to share but a "Unavailable" climate card in my HA dashboard

**Environment details:**
Host Operating System | Home Assistant OS 10.0
Update Channel | stable
Supervisor Version | supervisor-2023.04.1
Agent Version | 1.5.1
Docker Version | 23.0.3
Disk Total | 30.8 GB
Disk Used | 6.7 GB
Healthy | true
Supported | true
Board | ova
Supervisor API | ok
Version API | ok

 - Component version installed: latest from git 5/4/23
 - Last know working version: Version dated last October 2022
 - LG device type and model with issue: LG AC unit
 - LG devices connected (list): none at the moment

**Output of HA logs**
Paste the relavant output of the HA log here.

```
Logger: custom_components.smartthinq_sensors
Source: custom_components/smartthinq_sensors/wideq/core_async.py:380
Integration: SmartThinQ LGE Sensors ([documentation](https://github.com/ollo69/ha-smartthinq-sensors), [issues](https://github.com/ollo69/ha-smartthinq-sensors/issues))
First occurred: 4:31:52 PM (1 occurrences)
Last logged: 4:31:52 PM
Connection not available. ThinQ platform not ready

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1076, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1060, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 969, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 501, in sock_connect
    return await fut
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 541, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('13.209.172.37', 46030)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 242, in async_setup_entry
    client = await lge_auth.create_client_from_token(refresh_token, oauth2_url)
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 143, in create_client_from_token
    return await ClientAsync.from_token(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1436, in from_token
    await client.refresh()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1354, in refresh
    self._auth = await self.auth.refresh(True)
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 931, in refresh
    access_token, token_validity = await self._gateway.core.refresh_auth(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 649, in refresh_auth
    out = await self._auth_request(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 595, in _auth_request
    oauth_url = await self.get_oauth_url()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 380, in get_oauth_url
    async with self._get_session().post(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host kic.lgthinq.com:46030 ssl:default [Connect call failed ('13.209.172.37', 46030)]
```

**Additional context**
I am running HA in a proxmox server, so I can reset to a previous snapshot to try things.

**Comment by igorsantos07** [comment]
dup of #534 - it's always worthy to search before opening new issues :)

**Comment by rrediske** [comment]
My apologies, I should have mentioned that I did see issue #534, but I thought since I got so many extra python errors than issue #534, it was possibly a different problem.

I also tried to update this smartthinq integration using git clone, as described in the issue I opened on the HA git: [92530](https://github.com/home-assistant/core/issues/92530) referenced in my issue here, but that didn't work either. I might have done something wrong.

It looks like I can add the integration in the HA web gui now. Would this be the best way to proceed?

1. Shutdown HA.
2. Snapshot the working setup I have with proxmox.
3. Boot HA.
4. Remove the smartthinq integration in the gui.
5. Remove the smartthinq directory from the config folder in HA (in the shell, ala ssh addon)
6. Reboot HA.
7. Add the smartthinq integration from the HA web gui.
8. Update HA Core and System (which order is best, core then system or vice versa?)

**Comment by rrediske** [comment]
I revised my steps to remove automations (probably unnecessary, but "cleaner") and do HA updates before trying to bring smartthinq back, so:

1. Shutdown HA.
2. Snapshot the working setup I have with proxmox.
3. Boot HA.
4. Remove all automations referring to the LGE integration.
5. Remove the smartthinq integration in the gui.
6. Remove the smartthinq directory from the config/custom_integrations folder in HA (via ssh shell addon)
7. Reboot HA.
8. Update HA Core.
9. Update HA System
10. Add the smartthinq integration from the HA web gui.

The core update went to version 2023.05.02 already. I guess 2023.05.01 only lasted a few days before being replaced. I updated the system as well and did not get any errors, so I shelled in, did a git clone, copied the directory as documented in README.md, rebooted, added the integration and now I get the errors below.

I should mention that I can access my air conditioner on my LAN; ping etc, and the LG SmartThinQ android and iOS apps can both see and control the device as well. It might be worth mentioning that the apps are not complaining about agreeing to terms of service (I've had that hurdle twice in the last 18 months).

```
Logger: frontend.js.latest.202305033
Source: components/system_log/__init__.py:257
First occurred: 8:32:01 PM (1 occurrences)
Last logged: 8:32:01 PM

:0:0 ResizeObserver loop completed with undelivered notifications.
```
and
```
This error originated from a custom integration.

Logger: custom_components.smartthinq_sensors
Source: custom_components/smartthinq_sensors/wideq/core_async.py:380
Integration: SmartThinQ LGE Sensors (documentation, issues)
First occurred: 8:32:36 PM (1 occurrences)
Last logged: 8:32:36 PM

Connection not available. ThinQ platform not ready
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1076, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1060, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 969, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 501, in sock_connect
    return await fut
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 541, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('13.124.207.252', 46030)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 242, in async_setup_entry
    client = await lge_auth.create_client_from_token(refresh_token, oauth2_url)
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 143, in create_client_from_token
    return await ClientAsync.from_token(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1436, in from_token
    await client.refresh()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 1354, in refresh
    self._auth = await self.auth.refresh(True)
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 931, in refresh
    access_token, token_validity = await self._gateway.core.refresh_auth(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 649, in refresh_auth
    out = await self._auth_request(
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 595, in _auth_request
    oauth_url = await self.get_oauth_url()
  File "/config/custom_components/smartthinq_sensors/wideq/core_async.py", line 380, in get_oauth_url
    async with self._get_session().post(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host kic.lgthinq.com:46030 ssl:default [Connect call failed ('13.124.207.252', 46030)]
```

**Comment by rrediske** [comment]
I believe I found the problem. After monitoring network traffic, I noticed HA trying to talk to lots of *.amazonaws.com hosts in Korea. I didn't personally verify each IP was from Korea, but maxmind has all of them in the KR_v4 feed for geoip region KR. I have a pfSense router using maxmind for geoip blocking.

Each time I saw HA reach out to a new AWS server, I added that /24 subnet to my allow list. After seven rounds of adding subnets and still getting errors, I gave up and allowed all Korean IP's just to see if that changed anything. The integration now works and I have upgraded HA core and system successfully.

For the curious, here are the IP's that I saw HA failing to connect to (I imagine there are probably more):
43.200.167.162/24 # LG AC?
13.209.220.191/24 # LG AC?
54.180.73.84/24 # LG AC?
13.125.42.214/24 # LG AC?
13.124.61.109/24 # LG AC?
13.125.100.128/24 # LG AC?
